Completed for initial publication (#1)
investsuite/investsuite-api-docs: docs/advanced_topics/collections.md

            "readable_by":[
                "U01F5XFY0S38KJYVHX0X3GJ4KWM",
                "UXXXXXXXXXXXXXXXXXXXXXXXXXX"
            ],
            "modifiable_by":[
                "U01F5XFY0S38KJYVHX0X3GJ4KWM",
                "UXXXXXXXXXXXXXXXXXXXXXXXXXX"
            ],
            "id":"U01F5XFY0S38KJYVHX0X3GJ4KWM",
            "creation_datetime":"2021-05-17T00:00:00+00:00",
            "version":1,
            "version_datetime":"2021-05-17T15:43:23.427412+00:00",
            "version_authored_by_user_id":"UXXXXXXXXXXXXXXXXXXXXXXXXXX",
            "deleted":false,
            "_embedded":{
                
            },
            "first_name":"Laurent",
            "last_name":"Sorber",
            "email":"laurent.sorber@gmail.com",
            "phone":null,
            "brokerage_user_id":null,
            "idp_user_id":"a8c8b808-bad9-4acb-babf-aae52488e468",
            "language":"en-US",
            "products":[
                
            ]
        }
        }
    }    
    ```

## Query parameters

Collection endpoints accept alongside the `embed` query parameter two parameters: limit and offset. 

Parameter | Description
--------- | -----------

> limit     | Example: `limit=50`) Aallows you to pass in the number of items to be returned in the results array of the response. The default collection response size is `20 items`. The maximum size is 100.

Reviewer: ```suggestion
limit     | Example: `limit=50` Allows you to pass in the number of items to be returned in the results array of the response. The default collection response size is `20 items`. The maximum size is 100.
```

offset    |  Example `offset=50` To be used in combination with the `limit` parameter, the offset defines the number of records that need to be skipped.
embed     | Example: `embed=field_name_1,field_name_2` Optional comma-separated list of field names for which the referenced entity is to be included in the _embedded object of the response. See [Embedding](embedding.md).

> search    | Example: `query=email+eq+'jane.doe@example.com` See below

Reviewer: ```suggestion
search    | Example: `query=email+eq+'jane.doe@example.com` See below.
```

## Search

The API provides a structural search and filtering mechanism for all entities. We already opted to work with a query string parameter `query=email+eq+'jane.doe@example.com` instead of a POST body for more efficient client-side caching and ease of use for testing. The query syntax is a mix of both [OData](https://www.odata.org/) and [Apache Lucene Query Parser Syntax](https://lucene.apache.org/core/2_9_4/queryparsersyntax.html).

=== "Request"

    ```HTTP hl_lines="1"
    GET /portfolios
        ?query=lastmodified+in+['20200101'+to+'20240101'] HTTP/1.1
    Host: public-api.uat.investsuite.com
    Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJ...
    ```

### Fundamental rules

> - Each entity can be queried by issuing a `GET` request e.g. `GET users/?query=email+eq+'jane*'`.

Reviewer: ```suggestion
- Every entity type can be queried by issuing a `GET` request e.g. `GET users/?query=email+eq+'jane*'`.
```

> - Spaces are substituted with a plus sign ( + ) because space is not a valid character in an URL.

Reviewer: The real requirement is that the query string should be URL encoded. Normally spaces would be transformed to `%20`, but for improved legibility we recommend using `+` instead (which is also a valid way to URL encode spaces).

> - String and date literals are always enclosed in single-quotes. 

Reviewer: I don't think this is a hard requirement: string literals only need to be enclosed in single quotes if they contain characters that are used by the parser.

> - Spaces in a string literal are also denoted with a plus sign ( + ).

Reviewer: You can drop this if you say the query must be URL encoded.

> - If you need the plus sign in a string literal, just escape it with a backslash like email+eq+'*luk\\+*'.

Reviewer: Can also be dropped.

### Operators

Number Operators

> - Exact match (e.g. `age+eq+1`)

Reviewer: In the examples I think we might need to make the distinction between non-URL encoded queries, and the URL encoded version of it. Right now you're showing the URL encoded version. Maybe that's fine, but if we go with that I would explain above we'll always talk about the URL encoded versions in the rest of the documentation.

Author: I rearranged the Operators chapter into a table with an example column showing the examples non-URL encoded

- LT (less than) (e.g. `age+lt+18`)
- LTE (less than or equal) (e.g. `age+le+18`)
- GT (greater than) (e.g. `age+gt+18`)

> - GTE (greater than or equal) (e.g. `age+ge+18`)

Reviewer: I don't think the abbreviations at the beginning of the line add much value here, because we use different operators in the query itself. Maybe something like:
```suggestion
- Greater than or equal to: e.g. `age+ge+18`
```

String Operators

- Exact match (e.g. email+eq+'kristof.leroux@investsuite.com')
- Prefix search (starts with) (e.g. email+eq+kristof*')
- Postfix search (ends with)  (e.g. email+eq+'*gmail.com')
- LIKE matching (contains) (e.g. email+eq+'*gmail*')

Date Operators

- Exact match (e.g. `last_modified+eq+'2020-09-11T09:03:53.721588+00:00'`) (with or without time, with or without timezone)
- RANGE (is in range) for dates 
    - without time (e.g. `lastmodified+in+['20200101'+to+'20210101']`)
    - with time (e.g. `lastmodified+in+['2020-09-11T09:03:53.721588'+to+'2020-11-11T09:03:53.721588']`)
    - with timezone (e.g. `lastmodified+in+['2020-09-11T09:03:53.721588+00:00'+to+'2020-11-11T09:03:53.721588+00:00']`)

Boolean Operators

- Boolean Queries (e.g. `archived+eq+true`)

List Operators

- IN (element of) (e.g `field+in+['value1',+'value2',+'value3']`)

Logical Operators:

- AND (e.g. `email+eq+kristof+and+age+gt+18`)

> !!! Hint

Reviewer: Does this render to something?

Author: Yes this creates a styled blockquote https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types
